crates/vm/levm/src/precompiles.rs

```diff
@@ -203,25 +108,10 @@ pub const G1_POINT_AT_INFINITY: [u8; 128] = [0_u8; 128];
 pub const G2_POINT_AT_INFINITY: [u8; 256] = [0_u8; 256];
 
 pub fn is_precompile(address: &Address, fork: Fork, vm_type: VMType) -> bool {
-    // Cancun specs is the only one that allows point evaluation precompile
-    if *address == POINT_EVALUATION_ADDRESS && fork < Fork::Cancun {
-        return false;
-    }
-    // Prague or newers forks should only use these precompiles
-    // https://eips.ethereum.org/EIPS/eip-2537
-    if PRECOMPILES_POST_CANCUN.contains(address) && fork < Fork::Prague {
-        return false;
-    }
-
-    // P256 verify Precompile only existed on L2 before Osaka
-    if fork < Fork::Osaka && matches!(vm_type, VMType::L1) && address == &P256_VERIFICATION_ADDRESS
-    {
-        return false;
-    }
-
-    PRECOMPILES.contains(address)
-        || PRECOMPILES_POST_CANCUN.contains(address)
-        || address == &P256_VERIFICATION_ADDRESS
+    (matches!(vm_type, VMType::L2) && *address == P256_VERIFICATION.address)
+        || precompiles_for_fork(fork)
+            .iter()
+            .any(|precompile| precompile.address == *address)
```

Comment:
We should benchmark this and check that it doesn't tank our performance. If it ends up being an issue, I'm sure we can pre-generate a lookup table for the precompiles or something.

cc @Arkenan

Reply:
Nvm. The old implementation was almost the same, so it shouldn't impact performance as long as we [avoid allocating in `precompiles_for_fork`](https://github.com/lambdaclass/ethrex/pull/4625#discussion_r2382599630).